packages/web/components/Landing/OnboardingGame/Chiev.tsx

```diff
@@ -0,0 +1,431 @@
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  Text,
+  Tooltip,
+  usePrefersReducedMotion,
+  VStack,
+} from '@metafam/ds';
+import { httpLink } from '@metafam/utils';
+import { animated, useSpring } from '@react-spring/web';
+import OctoBg from 'assets/baby_octo.png';
+import { LandingConnectButton } from 'components/Landing/LandingConnectButton';
+import { MetaLink } from 'components/Link';
+import { CONFIG } from 'config';
+import { useGame } from 'contexts/GameContext';
+import { Contract, providers } from 'ethers';
+import { useWeb3 } from 'lib/hooks';
+import { get } from 'lib/store';
+import {
+  Dispatch,
+  SetStateAction,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+import { shortenAddress } from 'utils/stringHelpers';
+
+import ABI from '../../../contracts/BulkDisbursableNFTs.abi';
+
+export interface IChievMetadata {
+  name: string;
+  description: string;
+  image: string;
+  animation_url?: string;
+}
+
+export const Chiev = ({
+  won,
+  setWon,
+}: {
+  won: boolean;
+  setWon: Dispatch<SetStateAction<boolean>>;
+}) => {
+  const prefersReducedMotion = usePrefersReducedMotion();
+  const [noMotion, setNoMotion] = useState(false);
+  const root = typeof window !== 'undefined' ? document.body : null;
+  const { mintChiev, txLoading } = useGame();
+  const [claiming, setClaiming] = useState(false);
+  const { address: account, chainId, connected } = useWeb3();
+  const [wrongNetwork, setWrongNetwork] = useState(false);
+  const { alchemyApiKey, chievContractAddress: contractAddress } = CONFIG;
+  const claimed = get('ChievClaimed') === 'true' ?? false;
+  const chievId = BigInt(
+    '0x480000000000000000000000000000000000000000000000000000000002',
+  );
+  const [chievData, setChievData] = useState<IChievMetadata>();
+  // const [claimed, setClaimed] = useState(false);
+  // const toast = useToast();
+  const springProps = useSpring({
+    config: {
+      tension: 150,
+      friction: 10,
+      mass: 1,
+    },
+    // opacity: won ? 1 : 0,
+    transform: won ? 'translateY(0)' : 'translateY(-100%)',
+    scale: won ? 1 : 0.5,
+    display: won ? 'flex' : 'none',
+    loop: false,
+    immediate: noMotion,
+  });
+
+  const octoSpringProps = useSpring({
+    config: {
+      tension: 50,
+      friction: 10,
+      mass: 1,
+      delay: 750,
+    },
+    scale: won ? 1 : 0.7,
+    transformOrigin: 'center',
+    transform: won
+      ? 'translate3d(0, 0, 0) rotate(45deg)'
+      : 'translate3d(-100%, 100%, 0) rotate(40deg)',
+    immediate: noMotion,
+  });
+
+  const handleMinting = useCallback(async () => {
+    try {
+      setClaiming(true);
+      const tx = await mintChiev(chievId);
+      if (tx) {
+        console.log('receipt', tx);
+      }
+      setClaiming(false);
+    } catch (error: any) {
+      console.log('handleMintingError', error);
+    }
+  }, [chievId, mintChiev]);
+
+  /** function to get the token image url from the nft metadata */
+  const getTokenImage =
```

Comment:
We can move this function "outside" the component else this function will be recreated on every render of this component. 

I also noticed that all the information for this token (`alchemyApiKey`, `contractAddress`, `ABI` and the `cheivId`) are constant. In this case we can probably get the metadata when the server is statitically generated this page. 

What say @dysbulic ?